### PR TITLE
Study merge velocity for organization

### PR DESCRIPTION
**Pull request by thibaultduponchelle**
Hello,

We can totally provide all urls with multiple `--url` parameters but with this new option it is much more convenient.

```bash
$ perl -Ilib bin/github-mergevelocity --org=libwww-perl
.-------------+----------------------+----------+-----+----------+---------------+----------+---------------+----------+-----------------.
| user        | repo                 | velocity | PRs | merged   | merge days    | closed   | close days    | open     | open days       |
+-------------+----------------------+----------+-----+----------+---------------+----------+---------------+----------+-----------------+
| libwww-perl | URI-Nested           | 237      | 1   | 100% (1) | 1/PR (1)      | 0        | 0             | 0        | 0               |
| libwww-perl | LWP-MediaTypes       | 209      | 4   | 50% (2)  | 3/PR (6)      | 50% (2)  | 1/PR (2)      | 0        | 0               |
| libwww-perl | WWW-Mechanize-Cached | 168      | 20  | 90% (18) | 4/PR (72)     | 10% (2)  | 321/PR (642)  | 0        | 0               |
| libwww-perl | HTTP-Daemon          | 90       | 22  | 77% (17) | 4/PR (73)     | 18% (4)  | 264/PR (1057) | 5% (1)   | 1133/PR (1133)  |
| libwww-perl | WWW-Mechanize        | 83       | 100 | 90% (90) | 10/PR (886)   | 5% (5)   | 189/PR (947)  | 5% (5)   | 1694/PR (8469)  |
| libwww-perl | libwww-perl          | 78       | 100 | 76% (76) | 20/PR (1487)  | 17% (17) | 133/PR (2260) | 7% (7)   | 1070/PR (7491)  |
| libwww-perl | HTML-Parser          | 48       | 6   | 67% (4)  | 14/PR (55)    | 17% (1)  | 0             | 17% (1)  | 546/PR (546)    |
| libwww-perl | www-robotrules       | 0        | 0   | 0        | 0             | 0        | 0             | 0        | 0               |
| libwww-perl | HTML-Form            | 0        | 0   | 0        | 0             | 0        | 0             | 0        | 0               |
| libwww-perl | LWP-Protocol-GHTTP   | 0        | 0   | 0        | 0             | 0        | 0             | 0        | 0               |
| libwww-perl | HTTP-Message         | 0        | 0   | 0        | 0             | 0        | 0             | 0        | 0               |
| libwww-perl | HTTP-Date            | 0        | 0   | 0        | 0             | 0        | 0             | 0        | 0               |
| libwww-perl | Net-HTTP             | -9       | 38  | 71% (27) | 81/PR (2200)  | 26% (10) | 353/PR (3534) | 3% (1)   | 35/PR (35)      |
| libwww-perl | URI-db               | -71      | 12  | 67% (8)  | 0/PR (3)      | 17% (2)  | 2/PR (4)      | 17% (2)  | 1739/PR (3478)  |
| libwww-perl | HTTP-Cookies         | -239     | 40  | 85% (34) | 240/PR (8165) | 8% (3)   | 586/PR (1757) | 8% (3)   | 2088/PR (6264)  |
| libwww-perl | LWP-Protocol-https   | -257     | 38  | 61% (23) | 89/PR (2042)  | 29% (11) | 392/PR (4310) | 11% (4)  | 2211/PR (8843)  |
| libwww-perl | URI                  | -402     | 56  | 38% (21) | 13/PR (278)   | 36% (20) | 168/PR (3359) | 27% (15) | 1750/PR (26246) |
'-------------+----------------------+----------+-----+----------+---------------+----------+---------------+----------+-----------------'
```

Beware of API limits, it seems to eat it very fast.

### CODE REVIEW

File: lib/GitHub/MergeVelocity.pm [hunk 1]
```diff
@@ -55,11 +56,20 @@ option url => (
...
         'Full Github repo url or shorthand of username/repository.  You can pass multiple url args.',
 );
 
+option org => (
+    is       => 'ro',
+    isa      => ArrayRef,
+    format   => 's@',
+    required => 0,
+    documentation =>
+        'An organization.  You can pass multiple url args.',
```
Comment (oalders) [inline lib/GitHub/MergeVelocity.pm]: I think this could be on a single line. Did you run `perltidy` on this?

File: lib/GitHub/MergeVelocity.pm [hunk 2]
```diff
@@ -134,13 +144,45 @@ sub _build_report {
...
+    }
+
+    if ( $self->url ) {
+        push @urls, @{ $self->url };
+    }
+
+    # clean dups
+    my %seen = ();
+    @urls = grep { ! $seen{ $_ }++ } @urls;
+
```
Comment (oalders) [inline lib/GitHub/MergeVelocity.pm]: I think we could probably just simplify the above to:
```perl
use List::Util qw( uniq );

...

my @urls = uniq( @org_urls, @{ $self->url } );
```

File: lib/GitHub/MergeVelocity.pm [hunk 3]
```diff
@@ -165,6 +207,10 @@ sub print_report {
 
     my @repos = map { $self->_repository_for_url($_) } $self->_report_urls;
 
+    if ( ! @repos ) {
+	    return;
+    }
+
```
Comment (oalders) [inline lib/GitHub/MergeVelocity.pm]: This could also be `return unless @repos;`

**Review by oalders — CHANGES_REQUESTED** [review]
Thanks so much for this! Firstly, it's a great idea and secondly it makes me rethink how well I thought things were going with `libwww-perl`. 😆 

I don't see any real blockers here, but I did have a couple of suggestions which you could either consider or tell me that you can't be bothered. Your choice. 😄

File: lib/GitHub/MergeVelocity.pm (hunk 1 above)
Comment (thibaultduponchelle) [reply lib/GitHub/MergeVelocity.pm]: No I didn't

File: lib/GitHub/MergeVelocity.pm (hunk 2 above)
Comment (thibaultduponchelle) [reply lib/GitHub/MergeVelocity.pm]: Ok will change it 👍🏼

File: lib/GitHub/MergeVelocity.pm (hunk 3 above)
Comment (thibaultduponchelle) [reply lib/GitHub/MergeVelocity.pm]: Good point

File: lib/GitHub/MergeVelocity.pm (hunk 1 above)
Comment (oalders) [reply lib/GitHub/MergeVelocity.pm]: Might be handy to do in this case. An upcoming merge will enforce `perltidy`. :)

Comment (thibaultduponchelle) [reply lib/GitHub/MergeVelocity.pm]: ✅

File: lib/GitHub/MergeVelocity.pm (hunk 3 above)
Comment (thibaultduponchelle) [reply lib/GitHub/MergeVelocity.pm]: ✅

**Comment by thibaultduponchelle** [comment]
FYI `dzil build` gave me more updates that are not included in this PR (`cpanfile`, `Makefile.PL`, `META.json`, `README.md`)

File: bin/github-mergevelocity [hunk 4]
```diff
@@ -35,6 +35,12 @@ repo in it.  Be creative!
 
     github-mergevelocity --url https://github.com/neilbowers/PAUSE-Packages/issues
 
+=head2 org
+
+Expects an Github organization.  Mutliple C<--org> args can be supplied.
```
Comment (oalders) [inline bin/github-mergevelocity]: s/Mutliple/Multiple/

Comment (thibaultduponchelle) [reply bin/github-mergevelocity]: 👍🏼

**Review by oalders — APPROVED** [review]
Thanks for the changes! 💯

**Pull request merged**